pkg/network/websocket/hijack_test.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package websocket
+
+import (
+	"bufio"
+	"bytes"
+	"net"
+	"net/http"
+	"testing"
+)
+
+// TODO(tcnghia): Move this to knative/pkg.
+
+// hijackable is a http.ResponseWriter that implements http.Hijacker
+type hijackable struct {
+	http.ResponseWriter
+	w *bufio.ReadWriter
```

Comment:
does this have to be `*bufio.ReadWriter` or just `io.ReadWriter`?

Reply:
I use that in the return value of Hijack() to test the plumbing, so have to be `bufio.ReadWriter`